Sources/Confidence/Confidence.swift

```diff
@@ -1,45 +1,70 @@
 import Foundation
 
 public class Confidence: ConfidenceEventSender {
-    public var context: ConfidenceStruct
+    private let parent: ConfidenceContextProvider?
+    private var context: ConfidenceStruct
     public let clientSecret: String
     public var timeout: TimeInterval
     public var region: ConfidenceRegion
     public var initializationStrategy: InitializationStrategy
+    private var removedContextKeys: Set<String> = Set()
 
-    init(
+    required public init(
         clientSecret: String,
         timeout: TimeInterval,
         region: ConfidenceRegion,
-        initializationStrategy: InitializationStrategy
+        initializationStrategy: InitializationStrategy,
+        context: ConfidenceStruct = [:],
+        parent: ConfidenceEventSender? = nil
     ) {
-        self.context = [:]
         self.clientSecret = clientSecret
         self.timeout = timeout
         self.region = region
         self.initializationStrategy = initializationStrategy
+        self.context = context
+        self.parent = parent
     }
 
     // TODO: Implement actual event uploading to the backend
     public func send(definition: String, payload: ConfidenceStruct) {
         print("Sending \(definition) - Targeting key: \(payload)")
     }
 
+
+    public func getContext() -> ConfidenceStruct {
+        let parentContext = parent?.getContext() ?? [:]
+        var reconciledCtx = parentContext.filter {
+            !removedContextKeys.contains($0.key)
+        }
+        self.context.forEach { entry in
+            reconciledCtx.updateValue(entry.value, forKey: entry.key)
+        }
+        return reconciledCtx
+    }
+
     public func updateContextEntry(key: String, value: ConfidenceValue) {
         context[key] = value
+        removedContextKeys.remove(key)
     }
 
     public func removeContextEntry(key: String) {
         context.removeValue(forKey: key)
+        removedContextKeys.insert(key)
     }
 
     public func clearContext() {
+        removedContextKeys.removeAll()
```

Comment:
Not sure if this is the desired behaviour or if we intend to keep the removed keys

Reply:
I took inspiration from the Android provider and removed this API completely